index.js

```diff
@@ -3,89 +3,115 @@
 // # S3 storage module for Ghost blog http://ghost.org/
 var fs = require('fs');
 var path = require('path');
-var nodefn = require('when/node/function');
-var when = require('when');
-var readFile = nodefn.lift(fs.readFile);
-var unlink = nodefn.lift(fs.unlink);
-var AWS = require('aws-sdk');
+var Bluebird = require('bluebird');
+var AWS = require('aws-sdk-promise');
+var moment = require('moment');
+var readFileAsync = Bluebird.promisify(fs.readFile);
 var options = {};
 
 function S3Store(config) {
-  options = config || {};
+    options = config;
```

Comment:
I think there's a bug from giving options a default value. `{}` is truthy as shown by 
```javascript
if ({}) console.log('truthy')
// truthy
```